### PR TITLE
Correct the host port when using http health checks

### PR DESCRIPTION
**Pull request by aquamatthias**

### CODE REVIEW

**Comment by aquamatthias** [comment]
Cherry picked the commit from #4502 and fixed the lookup

**Comment by jdef** [comment]
Looks like a couple of TODOs were wiped out by accident. Health checks should work with container ports when the pod is running on a container network.

File: src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala [hunk 1]
```diff
@@ -342,14 +341,20 @@ object TaskGroupBuilder {
...
+    hostPorts: Seq[Option[Int]]): mesos.HealthCheck.Builder = {
+
     val builder = mesos.HealthCheck.newBuilder
     builder.setDelaySeconds(healthCheck.delaySeconds.toDouble)
     builder.setGracePeriodSeconds(healthCheck.gracePeriodSeconds.toDouble)
     builder.setIntervalSeconds(healthCheck.intervalSeconds.toDouble)
     builder.setConsecutiveFailures(healthCheck.maxConsecutiveFailures)
     builder.setTimeoutSeconds(healthCheck.timeoutSeconds.toDouble)
 
+    lazy val hostPortsByEndpoint: Map[String, Option[Int]] = {
```
Comment (jdef) [inline src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala]: worth an `assume` to check that hostPorts and the collection of endpoint names are the same size?

**Comment by jdef** [comment]
minor comment. otherwise lgtm

**Pull request merged**